configure.ac

```diff
@@ -35,12 +35,30 @@ PRESET_LDFLAGS="$LDFLAGS"
 dnl Set default LDFLAGS
 AS_IF([test "x$LDFLAGS" = "x"],[LDFLAGS="-g"])
 
-# Check for GNU cc
+# check for host OS detail
+AC_CANONICAL_HOST
+AC_MSG_CHECKING([simplified host os])
+simple_host_os=`echo $host_os|sed 's/[0-9].*//g;s/-.*//g'`
+squid_host_os_version=`echo $host_os|tr -d "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-"`
+AS_IF([test -n "$squid_host_os_version"],[
+  squid_host_os="`echo $simple_host_os| sed s/$squid_host_os_version//g`"
+],[
+  squid_host_os="$simple_host_os"
+])
+AC_MSG_RESULT($squid_host_os (version $squid_host_os_version))
+# on windows squid_host_os is either mingw or cygwin, version is 32
+
+# Find available compilers
 AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_CXX
 AC_LANG([C++])
```

Comment:
> AC_LANG (language):  Do compilation tests using the compiler, preprocessor, and file extensions for the specified language. 

AFAICT, the position of the AC_LANG macro contradicts AC_USE_SYSTEM_EXTENSIONS documentation: "[AC_USE_SYSTEM_EXTENSIONS] should be called _before_ any macros that run the C compiler".

Please consider moving AC_LANG after AC_USE_SYSTEM_EXTENSIONS.

Reply:
Done in [192671e](https://github.com/squid-cache/squid/pull/1139/commits/192671eae9d9f9ce0240b53a95b005743632f1c3)